### DISCUSSION TITLE
Default value combined valueAsDate broken after version upgrade

### DISCUSSION THREAD

**Discussion by miko55** [react-hook-form/react-hook-form#9566] [Q&A] [answered]
I have a form where I am setting default date to current one. I am using Calendar component from PrimeReact where I have added register - valueAsDate. It worked fine till version 7.39.3. That's last one where it's working fine with code that I currently have. After later version I get "Maximum update depth exceeded" ERROR.

I am pretty much begginer with React so before stating this as an issue I would like to check if I am doing something wrong. I have minimized code as much as I could:

`

    import { Controller, useForm } from 'react-hook-form';
    import { Calendar } from 'primereact/calendar';
    
    const CreateTransaction = () => {
    const defaultValues = {
    date: new Date(),
    };
    
    const { register, control } = useForm({ criteriaMode: 'all', defaultValues, mode: 'all' });
    
    return (
        <form>
            <div className="field col-4 xl:col-2">
                <label htmlFor="date" className="block">
                    Date*
                </label>
                <Controller
                    name="date"
                    control={control}
                    render={({ field }) => (
                        <Calendar
                            id={field.name}
                            {...field}
                            dateFormat="dd.mm.yy"
                            {...register('date', {
                                valueAsDate: true,
                            })}
                        />
                    )}
                />
            </div>
        </form>
    );
    };
    
    export default CreateTransaction;
`
As I said, this works on version 7.39.3 and stops working after upgrading to 7.39.4 or later. Is this error for an issue or am I doing something wrong ?

**Comment by Moshyfawn** [comment, marked as answer]
You are mixing 2 different APIs _(`register` and `Controller`)_ together, which can most possibly be the cause of your issue. The `register` method mostly used for uncontrolled native HTML inputs where the `Controller` is.. well, for use-cases where you need to control the value _(controlled inputs)_

**Reply by Moshyfawn** [reply]
Additional, your custom `Calendar` component already returns value as `date`, so you don't need to convert it to date with `valueAsDate: true` prop.

Here's a minimal example of RHF + `primereact/calender` usage:
```jsx
<Controller
  control={control}
  name="date"
  render={({ field: { ref, ...field } }) => (
    <Calendar {...field} inputRef={ref} />
  )}
/>
```

**Reply by miko55** [reply]
Thank you very much for explanation. I suspected that I am doing something wrong. I have check uncontrolled and controled input meaning and you are right about date return value from that component. I have fixed it now. Thank you one more time ;)
